code/game/objects/items/devices/radio/radio.dm

```diff
@@ -393,26 +394,32 @@ GLOBAL_LIST_INIT(default_medbay_channels, list(
 	if(ishuman(M))
 		var/mob/living/carbon/human/H = M
 		jobname = H.get_assignment()
+		rankname = H.get_authentification_rank()
 
 	// --- Carbon Nonhuman ---
 	else if(iscarbon(M)) // Nonhuman carbon mob
 		jobname = "No id"
+		rankname = "No id"
 
 	// --- AI ---
 	else if(isAI(M))
 		jobname = "AI"
+		rankname = "AI"
 
 	// --- Cyborg ---
 	else if(isrobot(M))
 		jobname = "Cyborg"
+		rankname = "Cyborg"
 
 	// --- Personal AI (pAI) ---
 	else if(ispAI(M))
 		jobname = "Personal AI"
+		rankname = "Personal AI"
 
 	// --- Unidentifiable mob ---
 	else
 		jobname = "Unknown"
+		rankname = "Unknown"
```

Comment:
These overrides shouldnt be needed given it will always have a `jobname`

Reply:
I am not on my coding machine right now so I can't test.
But rankname isn't declared without an initial value, and I am unsure how that would go along with the assignment to tcm.sender_rank (which is declared with the initial value of "error")

If it doesn't overwrite the existing sender_rank, then it will remain on display (for anyone with access to view variables) as "error", which isn't true since there is no error,
And if it does overwrite, might throw a runtime if it does assign to an empty value, and then we try to use this value for any other operation later down the line. 

Regardless, you are more knowledgeable on the inner workings of BYOND than me, I'll remove it if you think it's unnecessary.